source/gui/settingsDialogs.py

```diff
@@ -928,144 +928,193 @@ def onOk(self, evt):
 			self.Parent.updateCurrentSynth()
 		super(SynthesizerSelectionDialog, self).onOk(evt)
 
-class SynthSettingChanger(object):
+class DriverSettingChanger(object):
 	"""Functor which acts as calback for GUI events."""
 
-	def __init__(self,setting):
+	def __init__(self,driver,setting):
+		self._driverRef=weakref.ref(driver)
 		self.setting=setting
 
+	@property
+	def driver(self):
+		return self._driverRef()
+
 	def __call__(self,evt):
 		val=evt.GetSelection()
-		setattr(getSynth(),self.setting.name,val)
+		setattr(self.driver,self.setting.name,val)
 
-class StringSynthSettingChanger(SynthSettingChanger):
-	"""Same as L{SynthSettingChanger} but handles combobox events."""
-	def __init__(self,setting,panel):
-		self.panel=panel
-		super(StringSynthSettingChanger,self).__init__(setting)
+class StringDriverSettingChanger(DriverSettingChanger):
+	"""Same as L{DriverSettingChanger} but handles combobox events."""
+	def __init__(self,driver,setting,container):
+		self.container=container
+		super(StringDriverSettingChanger,self).__init__(driver,setting)
 
 	def __call__(self,evt):
+		# Quick workaround to deal with voice changes.

```

Comment:
Since voice has special behavior, I would prefer to see a `VoiceDriverSettingChanger` subclass. Then the check for the `voice` setting name can be used to create this subclass.

Reply:
I agree that this is still hacky. However this doesn't only mean that we need a subclass of StringDriverSettingChanger. There is also makeSliderSettingControl that needs to know about the different functor for voice.